gc/base/standard/Scavenger.cpp

```diff
@@ -1505,7 +1507,13 @@ MM_Scavenger::copyObject(MM_EnvironmentStandard *env, MM_ForwardedHeader* forwar
 //		return NULL;
 //	}
 #endif /* OMR_GC_CONCURRENT_SCAVENGER */
-	return copy(env, forwardedHeader);
+	omrobjectptr_t result = NULL;
+	if (IS_CONCURRENT_ENABLED) {
+		result = MM_Scavenger::copy<true>(env, forwardedHeader);
+	} else {
+		result = MM_Scavenger::copy<false>(env, forwardedHeader);
+	}
+	return result;
```

Comment:
This pattern repeats twice, so we could put it in another inline method. We can call this one plain _copy_, so that the 2 call sites would not change ,at all. The original _copy_ could be renamed to something more specific. like copyVersioned or copyForVersion, copyForVariant or something else if you have a good suggestion.

To make it really nice to read, the parameter of copyForVersion could be enum {CS, STW}, and then the call would look like copyForVariant<CS>() etc. That would make laer _if_ checks a bit more ugly - instead of just checking bool csEnabled, we would check (variant == CS)

Reply:
lmk how it looks

Reply:
Looks good, we can go ahead with this.
Still, while it's in build pipeline, you could verify the gain is still there (unlikely, but enum or some other small thing could have messed things up).

Reply:
update commit/PR comments